src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -191,16 +191,44 @@ public void Initialize(string fileName)
             {
                 // failed to find the VS Service
             }
-            string natvisPath = null;
-            string directory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
-            if (!string.IsNullOrEmpty(directory) && !string.IsNullOrEmpty(fileName))
+
+            if (string.IsNullOrEmpty(Path.GetDirectoryName(fileName)))
             {
-                natvisPath = Path.Combine(directory, fileName);
-            }
+                string globalVisualizersDirectory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
+                string globalNatVisPath = null;
+                if (!string.IsNullOrEmpty(globalVisualizersDirectory) && !string.IsNullOrEmpty(fileName))
+                {
+                    globalNatVisPath = Path.Combine(globalVisualizersDirectory, fileName);
+                }
 
-            if (!string.IsNullOrEmpty(natvisPath))
+                // For local launch, try and load natvis next to the target exe if it exists.
+                if (_process.LaunchOptions is LocalLaunchOptions)
+                {
+                    string natvisPath = Path.Combine(Path.GetDirectoryName((_process.LaunchOptions as LocalLaunchOptions).ExePath), fileName);
```

Comment:
Unrooted exe shouldn't be possible, but this may be called before that is validated. I added a check just to be safe.